packages/cli/src/controller/migrate/manifest/migrate-manifest.controller.ts

```diff
@@ -0,0 +1,144 @@
+    throw new Error(`All network values in subgraph Networks should be the same. Got ${subgraphDsNetworks}`);
+  }
+  const firstDsKind = subgraphDsKinds[0];
+  const firstDsNetwork = subgraphDsNetworks[0];
+  const networkFamily = graphToSubqlNetworkFamily[firstDsKind];
+  if (!networkFamily) {
+    throw new Error(`Corresponding subquery network is not find with subgraph data source kind ${firstDsKind}`);
+  }
+  return {networkFamily, chainId: getChainIdByNetworkName(networkFamily, firstDsNetwork)};
+}
+
+/**
+ * Render subquery project to .ts file
+ * @param projectPath
+ * @param project
+ */
+export async function renderManifest(projectPath: string, project: CommonSubqueryProject): Promise<void> {
+  try {
+    await renderTemplate(PROJECT_TEMPLATE_PATH, projectPath, {
+      props: {
+        importTypes: {
+          network: 'ethereum',
+          projectClass: 'EthereumProject',
+          projectDatasourceKind: 'EthereumDatasourceKind',
+          projectHandlerKind: 'EthereumHandlerKind',
+        },
+        projectJson: project,
+      },
+      helper: {
+        upperFirst,
+      },
+    });
+  } catch (e) {
+    throw new Error(`Failed to create project manifest, ${e}`);
+  }
+}
+
+/**
+ *  Migrate a subgraph project manifest to subquery manifest file
+ * @param network network family
+ * @param inputPath file path to subgraph.yaml
+ * @param outputPath file path to project.ts
+ */
+export async function migrateManifest(chainInfo: ChainInfo, subgraphManifest: SubgraphProject, outputPath: string) {
+  await prepareDirPath(outputPath, false);
+  await renderManifest(outputPath, graphManifestToSubqlManifest(chainInfo, subgraphManifest));
+}
+
+/**
+ * Convert the graph project to subquery project
+ * @param network
+ * @param subgraphManifest
+ */
+function graphManifestToSubqlManifest(chainInfo: ChainInfo, subgraphManifest: SubgraphProject): CommonSubqueryProject {
+  return {
+    network: {chainId: chainInfo.chainId, endpoint: ''},
+    name: subgraphManifest.name,
+    specVersion: '1.0.0',
+    runner: {
+      node: {version: '^', name: findRunnerByNetworkFamily(chainInfo.networkFamily)},
+      query: {version: '^', name: '@subql/query'},
+    },
+    version: subgraphManifest.specVersion,
+    dataSources: subgraphDsToSubqlDs(chainInfo.networkFamily, subgraphManifest.dataSources),
+    description: subgraphManifest.description,
+    schema: subgraphManifest.schema,
+    repository: '',
+    templates: subgraphManifest.templates
+      ? subgraphTemplateToSubqlTemplate(chainInfo.networkFamily, subgraphManifest.templates)
+      : undefined,
+  };
+}
+
+export function subgraphTemplateToSubqlTemplate(
+  network: NETWORK_FAMILY,
+  subgraphTemplates: SubgraphTemplate[]
+): TemplateKind[] {
+  const convertFunction = networkConverters[network as NETWORK_FAMILY]?.templateConverter;
+  if (!convertFunction) {
+    throw new Error(`${network} does not support migration of templates.`);
+  }
```

Comment:
Checking the network exists can be moved to `graphManifestToSubqlManifest`. We don't need to check multiple times and `templateConverter` should be a requirement for each network family